### PR TITLE
feat(sdk-coin-ltc): refactor ltc to its own module

### PR DESCRIPTION
**Pull request by mmcshinsky-bitgo**
- Refactor Litecoin to its' own module (sdk-coin-ltc)
- Update generator for package installation and tsconfig settings
BG-51038

### CODE REVIEW

File: modules/sdk-coin-ltc/src/ltc.ts [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+} from '@bitgo/sdk-core';
 import * as utxolib from '@bitgo/utxo-lib';
 
+export interface VerifyRecoveryTransactionOptions extends BaseVerifyRecoveryTransactionOptions {
+  transactionHex: string;
+}
+
 export class Ltc extends AbstractUtxoCoin {
-  constructor(bitgo: BitGoBase, network?: UtxoNetwork) {
+  protected constructor(bitgo: BitGoBase, network?: UtxoNetwork) {
```
Comment (bitgoAaron) [inline modules/sdk-coin-ltc/src/ltc.ts:15]: this shouldn't be protected

File: modules/sdk-coin-ltc/src/tltc.ts [hunk 2]
```diff
@@ -17,11 +17,15 @@ export class Tltc extends Ltc {
     return new Tltc(bitgo);
   }
 
-  getChain() {
+  getChain(): string {
     return 'tltc';
   }
 
-  getFullName() {
+  getFamily(): string {
```
Comment (bitgoAaron) [inline modules/sdk-coin-ltc/src/tltc.ts:24]: family is environment agnostic, this should not override the base class.

File: scripts/sdk-coin-generator/index.js [hunk 3]
```diff
@@ -174,11 +175,30 @@ module.exports = class extends Generator {
     addNewCoinToBitgo(this.contextRoot, this.answers);
     addNewCoinToBitgoTsConfig(this.contextRoot, this.answers);
 
-    await this.addDependencies({
-      '@bitgo/sdk-core': '^1.0.1',
-    });
+    switch (this.answers.boilerplate) {
+      case 'utxo':
+        await this.addDependencies({
+          '@bitgo/abstract-utxo': '^1.0.0',
```
Comment (bitgoAaron) [inline scripts/sdk-coin-generator/index.js:181]: can these versions be resolved from the current package.json files instead of hard coding them?

**Review by bitgoAaron — CHANGES_REQUESTED** [review]
a few changes necessary

File: scripts/sdk-coin-generator/index.js (hunk 3 above)
Comment (mmcshinsky-bitgo) [reply scripts/sdk-coin-generator/index.js:181]: Yeah, I can go fetch those from the files.

File: modules/sdk-coin-ltc/test/unit/index.ts [hunk 4]
```diff
@@ -1,45 +1,46 @@
...
+import { BitGoAPI } from '@bitgo/sdk-api';
+
+describe('Litecoin:', function () {
+  const bitgo: TestBitGoAPI = TestBitGo.decorate(BitGoAPI, { env: 'test' });
+  bitgo.initializeTestVars();
+  bitgo.safeRegister('ltc', Ltc.createInstance);
+  bitgo.safeRegister('tltc', Tltc.createInstance);
+
+  const ltc = bitgo.coin('ltc') as Ltc;
+  const tltc = bitgo.coin('tltc') as Tltc;
```
Comment (OttoAllmendinger) [inline modules/sdk-coin-ltc/test/unit/index.ts:15]: see comment here https://github.com/BitGo/BitGoJS/pull/2440/files#r912627038

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]

File: modules/sdk-coin-ltc/test/unit/index.ts (hunk 4 above)
Comment (mmcshinsky-bitgo) [reply modules/sdk-coin-ltc/test/unit/index.ts:15]: I did a rebase so I might have removed what was there (if I had changed it previously locally). `getUtxoCoin` is in `bitgo` right now while these tests are in `sdk-coin-ltc`. If we do any refactoring and consolidating on the tests side, method movements will probably happen when we do the utxo test refactor for all the utxo packages.

**Review by OttoAllmendinger — DISMISSED** [review]

**Comment by bitgoAaron** [comment]
looks like tltc is still overriding getFamily incorrectly:
https://github.com/BitGo/BitGoJS/blob/d786d918a709bccc7e352b71c04fa6fe92516c38/modules/sdk-coin-ltc/src/tltc.ts#L21

**Review by bitgoAaron — CHANGES_REQUESTED** [review]
don't override getFamily in `tltc.ts`

**Review by bitgoAaron — APPROVED** [review]

**Pull request merged**